lib/streamlit/state/session_state.py

```diff
@@ -54,27 +198,29 @@ class SessionState(MutableMapping[str, Any]):
     """
 
     _old_state: Dict[str, Any] = attr.Factory(dict)
-    _new_state: Dict[str, Any] = attr.Factory(dict)
-
-    def make_state_old(self) -> None:
-        self._old_state.update(self._new_state)
-        self._new_state.clear()
+    _new_session_state: Dict[str, Any] = attr.Factory(dict)
+    _new_widget_state: WStates = attr.Factory(WStates)
+
+    # TODO: widgets being moved to the old state dict should be unwrapped
+    # is it possible for a value to get through this without being deserialized?
+    def _compact_state(self) -> None:
```

Comment:
nit: I think a good number of the `SessionState` methods shouldn't have the underscore prefix since they're used outside of this module. Looks like `_compact_state`, `_set_from_proto`, `_call_callbacks`, `_reset_triggers`, `_cull_nonexistent`, `_set_metadata`, `_get_value_for_registration`, and `_as_widget_states` are all used outside of this file so should likely have the prefix dropped

Reply:
I had somehow forgotten that users interact with only the stuff exposed by the Lazy wrapper, and was putting underscores to hide them from apps. Agreed that they should not have underscores as they are not intended to be private.